### ISSUE TITLE
Mod doesn't register players in the building

### ISSUE THREAD

**Issue by Dayson13** [ControlCompany/ControlCompany#184]
**Describe the bug**
The mod sometimes doesn't register the target player in the building while using ghost mode. While in ghost mode, it still spawn the outdoor monster while the target player is inside the building and also night vision is not working while having this bug.

**To Reproduce**

**Expected behavior**
This bug happens after I installed the "LC_Better_Teleport" mod which will remove the inverse teleporter cooldown and prevent players  dropping items after tp. I don't know if it is due to this mod or it is because players need to enter the building from the main entrance or the fire exit once if I want to spawn the indoor monster to the target player.

**Screenshots**
Will provide sceenshots later

**Mod version (please complete the following information):**
v1.1.8

**Additional context**

**Comment by Dayson13** [comment]
I am now sure it is not due to the mod "LC_Better_Teleport", the bug mostly occur when players used the inverse teleporter to enter the building.

**Comment by JustCallMeQuincy** [comment]
Thanks for testing! Have you tried using the inverse teleporter without this mod? I've personally not run into this issue (I never tested `LC_BeterTeleport`).

**Comment by Dayson13** [comment]
Yes, I have disabled the LC_BeterTeleport mod already and the bug still exist, maybe I should  try to disable other mods to see what happen.

**Comment by JustCallMeQuincy** [comment]
That's a good idea, do let us know what you find!
